.github/workflows/test_scripts.yml

```diff
@@ -19,12 +19,18 @@ jobs:
             python-cleanliness,
             astyle-cleanliness,
             validate_board_list,
+            pre-commit-cleanliness,
        ]
     steps:
       # git checkout the PR
       - uses: actions/checkout@v4
         with:
           submodules: 'recursive'
+      - name: install pre-commit
+        shell: bash
+        run: |
+          pre-commit install
```

Comment:
why need this ? we don't run git command on CI, so the install isn't need normally, or I miss something ?

Reply:
are you sure that shouldn't be `pip install pre-commit `instead ?

Reply:
OK i have checked and we got pre-commit into the docker image already so we are fine. the pre-commit install is just to install the git hook so that isn't need on CI

Reply:
You are conflating installation of the pre-commit tool in the environment and installation of the git hooks into the repo. 
https://pre-commit.com/#3-install-the-git-hook-scripts

We need both at some point before calling `pre-commit run` in ArduPilot.

Reply:
Nope you don't need to install the git hook to use precommit.